### PR TITLE
misc/multihash: Add Blake2b256 and Blake2s128

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

Add `Blake2b256` and `Blake2s128` hash types for `multihash`.

### CODE REVIEW

**Comment by ec2** [comment]
Is there a reason that libp2p doesnt use the multihash library? https://github.com/multiformats/rust-multihash

**Comment by tomaka** [comment]
> Is there a reason that libp2p doesnt use the multihash library? https://github.com/multiformats/rust-multihash

There was a time when we were frequently doing changes, and it would have been very slow to open individual PRs in the multihash repo and wait for them to be merged.
I think that nowadays we could switch back.

cc #759

**Comment by koushiro** [comment]
Could someone review this PR？

File: misc/multihash/src/lib.rs [hunk 1]
```diff
@@ -115,6 +118,80 @@ fn encode_hash(hash: Hash) -> (usize, BytesMut) {
     (code.len() + 1, output)
 }
 
+/// BLAKE2b-256 (32-byte hash size)
+#[derive(Debug, Clone)]
+struct Blake2b256(blake2::VarBlake2b);
+
+impl Default for Blake2b256 {
+    fn default() -> Self {
+        use digest::VariableOutput;
```
[the remark this answers is not in the document]
Comment (koushiro) [reply misc/multihash/src/lib.rs]: Okay, PTAL again :)

**Comment by koushiro** [comment]
@tomaka Can you release a new version (`parity-multihash v0.2.2`) with this change ?

**Pull request merged**